frontend/src/metabase/lib/data_grid.js

```diff
@@ -37,12 +37,17 @@ export function pivot(data) {
         normalColValues.sort();
     }
 
+
     // make sure that the first element in the pivoted column list is null which makes room for the label of the other column
     pivotColValues.unshift(data.cols[normalCol].display_name);
 
+    // add a total column, this needs to happen here so that we make space for the eventual total
+    pivotColValues = pivotColValues.concat(["Total"])
+
     // start with an empty grid that we'll fill with the appropriate values
-    const pivotedRows = normalColValues.map((normalColValues, index) => {
+    let pivotedRows = normalColValues.map((normalColValues, index) => {
         const row = pivotColValues.map(() => null);
```

Comment:
could this mapping method be in its own function?

Basically each `pivotedRows` transformation could be in its own mapping function, and the method call could be something like this:

```
const pivotedRows = normalColValues
    .map(createEmptyGrid)
    .map(populateWithData)
    .map(addTotalToRows)
    .map(addTotalToColumns)
```

This would conveniently also eliminate the mutation that makes reading the function confusing.